docs/reference/api-conventions.asciidoc

```diff
@@ -214,6 +214,37 @@ instead. When using this method, the `source_content_type` parameter
 should also be passed with a media type value that indicates the format
 of the source, such as `application/json`.
 
+[discrete]
+[[api-compatibility]]
+=== REST API version compatibility
+
+Major version upgrades often include a number of breaking changes 
+that impact how you interact with {es}. 
+While we recommend that you monitor the deprecation log and 
+update applications before upgrading {es},
+having to coordinate the necessary changes can be an impediment to upgrading. 
+
+To enable an application to continue to function without modification after an upgrade, 
+you can include API compatibility headers that tell {es} that 
+you are still using the previous version of the REST API. 
+Using the headers allows the structure of requests and responses to remain the same;
+it does not guarantee the same behavior. 
```

Comment:
```suggestion
You can enable an existing application to function without modification after
an upgrade by including API compatibility headers, which tell {es} you are still
using the previous version of the REST API. Using these headers allows the 
structure of requests and responses to remain the same; it does not guarantee
the same behavior. 
```

Reply:
I had to re-read the first sentence a few times to parse it. Could just be me, but I offered an alternative wording that might simplify readability.